### PR TITLE
Document the grade option

### PR DESCRIPTION
**Pull request by dholbach**
Document the `grade` option.
    
LP: #1630914

### CODE REVIEW

File: docs/snapcraft-syntax.md [hunk 1]
```diff
@@ -21,6 +21,11 @@ contain.
   The type of confinement supported by the snap. Can be either "devmode" (i.e.
   this snap doesn't support running under confinement) or "strict" (i.e. full
   confinement supported via interfaces).
+* `grade` (string)
+  This defines the quality grade of the snap. It can be either "devel" (i.e.
+  a development version of the snap, so not to be published to the "stable" or
+  "candidate" channels) or "stable" (i.e. a stable release or release
+  candidate).
```
Comment (sergiusens) [inline docs/snapcraft-syntax.md]: the _i.e. a stable or release candidate_ seems more to the point. But I am also afraid people will assume by reading this that a `grade: stable` snap cannot be released to the `beta` and `edge` channels.

Comment (dholbach) [inline docs/snapcraft-syntax.md]: Good point.

File: docs/upload-your-snap.md [hunk 2]
```diff
@@ -4,9 +4,24 @@ So you've been working hard on your snap, and you finally have it to the point
...
+`grade` in your `snapcraft.yaml` file can be either `stable` or `devel`. It
+essentially tags the resulting snap for life with the grade it should have.
+
+How is this useful? Imagine you are developing a snap and have your continuous
+integration setup to always push to the `edge` channel; in this case one would
+set `grade: devel`. This ensures that this snap is not accidentally published
+to the `stable` channels.
+
+A snap with `grade: devel` cannot be released to the `stable` or `candidate`
+channels.
```
Comment (sergiusens) [inline docs/upload-your-snap.md]: hey, this rings a bell :-P

File: docs/snapcraft-syntax.md [hunk 3]
```diff
@@ -21,6 +21,11 @@ contain.
   The type of confinement supported by the snap. Can be either "devmode" (i.e.
   this snap doesn't support running under confinement) or "strict" (i.e. full
   confinement supported via interfaces).
+* `grade` (string)
+  This defines the quality grade of the snap. It can be either "devel" (i.e.
+  a development version of the snap, so not to be published to the "stable" or
+  "candidate" channels) or "stable" (i.e. a stable release or release
+  candidate, which can be released to all channels though).
```
Comment (sergiusens) [inline docs/snapcraft-syntax.md]: I guess remove the "though" at the end.

File: docs/upload-your-snap.md [hunk 4]
```diff
@@ -38,7 +53,8 @@ Now, let's upload that snap!
 
 ## Upload the snap
 
-Run the `snapcraft upload` command passing the path to the `snap` file as an argument:
+Run the `snapcraft upload` command passing the path to the `snap` file as an
```
Comment (sergiusens) [inline docs/upload-your-snap.md]: hmm, this change seems out of place

File: docs/upload-your-snap.md [hunk 5]
```diff
@@ -4,9 +4,24 @@ So you've been working hard on your snap, and you finally have it to the point
...
+to the `stable` channels.
+
+A snap with `grade: devel` cannot be released to the `stable` or `candidate`
+channels.
+
+
 ## Build the snap
 
-Get into the directory containing the `snapcraft.yaml` file, and do the following:
+Get into the directory containing the `snapcraft.yaml` file, and do the
```
Comment (sergiusens) [inline docs/upload-your-snap.md]: hmm, this change seems out of place

File: docs/upload-your-snap.md [hunk 6]
```diff
@@ -49,7 +65,7 @@ Run the `snapcraft upload` command passing the path to the `snap` file as an arg
     Please check out the application at: https://myapps.developer.ubuntu.com/dev/click-apps/1337/
 
 Your snap has now been uploaded to the store, and is now undergoing an
-automated review. You'll be emailed when the review has completed, at which time
-you can visit the MyApps site and publish your snap! Note that if you uploaded
-a new version for an already-published snap, your update will be automatically
-published.
+automated review. You'll be emailed when the review has completed, at which
```
Comment (sergiusens) [inline docs/upload-your-snap.md]: hmm, this change seems out of place

File: docs/upload-your-snap.md (hunk 5 above)
Comment (dholbach) [inline docs/upload-your-snap.md]: I was just wrapping at 80 chars.

**Comment by sergiusens** [comment]
El 07/10/16 a las 03:04, Daniel Holbach escribió:
>
> *@dholbach* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In docs/upload-your-snap.md 
> <https://github.com/snapcore/snapcraft/pull/857>:
>
> >  ## Build the snap
>   
> -Get into the directory containing the `snapcraft.yaml` file, and do the following:
> +Get into the directory containing the `snapcraft.yaml` file, and do the
>
> I was just wrapping at 80 chars.
>

Can we revert that, it exposes others things I want to change in that 
doc anyways :-)

**Comment by dholbach** [comment]
Done.

**Pull request merged**